Lib/test/test_modulefinder.py

```diff
@@ -351,6 +353,20 @@ def test_replace_paths(self):
         output = output.getvalue()
         expected = "co_filename %r changed to %r" % (old_path, new_path)
         self.assertIn(expected, output)
+    
+    def test_encoding(self):
+        encoding_test = [
+            "a",
+            ["a", "b"],
+            [], [],
+            """\
+a.py
+                                # coding=latin-1
+                                import b
+b.py
+                                # coding=latin-1
+"""]
```

Comment:
I don't think this actually exercises your patch -- a proper test should fail before your patch and succeed afterwards

I believe what you need to test this is to write out a file with an exotic encoding and ensure that the modulefinder still succeeds on that.

For example:
```python
with open('f.py', 'w', encoding='cp1252') as f:
    f.write('import b\nx = "€"\b')
```

currently this will fail with a UnicodeDecodeError, but should succeed after your patch

Reply:
Thank you so much! I was a little lost while creating the tests.